InteroperabilityTest_Engine/Compute/PushPullCompare.cs

```diff
@@ -47,17 +48,16 @@ public static partial class Compute
         /**** Public Methods                            ****/
         /***************************************************/
 
-        [Description("Tests Pushing objects of a specific type, then pulling them back and comparing the objects. Returns Results outlining if the objects pulled are identical to the pushed ones, and if not, what properties are different between the two")]
-        [Input("adapter", "The instance of the adapter to test for")]
-        [Input("type", "The type of object to test. This will use test sets in the Dataset library")]
+        [Description("Tests Pushing objects of a specific type, then pulling them back and comparing the objects. Returns Results outlining if the objects pulled are identical to the pushed ones, and if not, what properties are different between the two.")]
+        [Input("adapter", "The instance of the adapter to test for.")]
+        [Input("type", "The type of object to test. This will use test sets in the Dataset library.")]
         [Input("config", "Config for the test. Controls whether the adapter should be reset between runs and what comparer to use.")]
-        [Input("active", "Toggles whether to run the test")]
-        [MultiOutput(0, "diffingResults", "Diffing results outlining any differences found between the pushed and pulled objects.")]
-        [MultiOutput(1, "adapterLog", "A list of any error or warning messages returned by the adapter in the process, as a dictionary sorted by name of the set.")]
-        public static Output<List<InputOutputComparison>, Dictionary<string, List<Event>>> PushPullCompare(BHoMAdapter adapter, Type type, PushPullCompareConfig config = null, bool active = false)
+        [Input("active", "Toggles whether to run the test.")]
+        [Output("testResult", "Diffing results outlining any differences found between the pushed and pulled objects.")]
+        public static List<TestResult> PushPullCompare(BHoMAdapter adapter, Type type, PushPullCompareConfig config = null, bool active = false)
         {
             if (!active)
-                return new Output<List<InputOutputComparison>, Dictionary<string, List<Event>>>();
+                return new List<TestResult>();
```

Comment:
Personally prefer empty lists over nulls when we can. When run from the UI (for example Grasshopper) this means you get nothing back, instead of a null that you would have to clean up